hist/hist/src/THnBase.cxx

```diff
@@ -83,6 +83,82 @@ THnBase::THnBase(const char *name, const char *title, Int_t dim, const Int_t *nb
    fAxes.SetOwner();
 }
 
+THnBase::THnBase(const THnBase &other)
+   : TNamed(other), fNdimensions(other.fNdimensions), fAxes(fNdimensions), fBrowsables(fNdimensions),
+     fEntries(other.fEntries), fTsumw(other.fTsumw), fTsumw2(other.fTsumw2), fTsumwx(other.fTsumwx),
+     fTsumwx2(other.fTsumwx2), fIntegral(other.fIntegral), fIntegralStatus(other.fIntegralStatus)
+{
+
+   for (Int_t i = 0; i < fNdimensions; ++i) {
+      TAxis *axis = new TAxis(*static_cast<TAxis *>(other.fAxes[i]));
+      fAxes.AddAtAndExpand(axis, i);
+   }
+   fAxes.SetOwner();
+}
+
+THnBase &THnBase::operator=(const THnBase &other)
+{
+
+   if (this == &other)
+      return *this;
+
+   TNamed::operator=(other);
+   fNdimensions = other.fNdimensions;
+   fAxes = TObjArray(fNdimensions);
+   fBrowsables = TObjArray(fNdimensions);
+   fEntries = other.fEntries;
+   fTsumw = other.fTsumw;
+   fTsumw2 = other.fTsumw2;
+   fTsumwx = other.fTsumwx;
+   fTsumwx2 = other.fTsumwx2;
+   fIntegral = other.fIntegral;
+   fIntegralStatus = other.fIntegralStatus;
+
+   for (Int_t i = 0; i < fNdimensions; ++i) {
+      TAxis *axis = new TAxis(*static_cast<TAxis *>(other.fAxes[i]));
+      fAxes.AddAtAndExpand(axis, i);
+   }
+   fAxes.SetOwner();
+
+   return *this;
+}
+
+THnBase::THnBase(THnBase &&other)
+   : TNamed(std::move(other)), fNdimensions(other.fNdimensions), fAxes(other.fAxes), fBrowsables(fNdimensions),
+     fEntries(other.fEntries), fTsumw(other.fTsumw), fTsumw2(other.fTsumw2), fTsumwx(std::move(other.fTsumwx)),
+     fTsumwx2(std::move(other.fTsumwx2)), fIntegral(std::move(other.fIntegral)), fIntegralStatus(other.fIntegralStatus)
+{
+
+   other.fAxes.SetOwner(false);
+   other.fAxes.Clear();
+   fAxes.SetOwner();
+}
+
+THnBase &THnBase::operator=(THnBase &&other)
+{
+
+   if (this == &other)
+      return *this;
+
+   TNamed::operator=(std::move(other));
```

Comment:
Shouldn't this technically be last? (as it technically leaves 'other' is a useable but invalid state?

Reply:
This ordering matches what the implicitly-defined move assignment operator would have done.  (And it should be ok because it's only the TNamed part of other which is left in an invalid state.)